src/testsuite/arith128_test_f64.c

```diff
@@ -0,0 +1,2230 @@
+ You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+
+ arith128_test_f64.c
+
+ Contributors:
+      IBM Corporation, Steven Munroe
+      Created on: Jul 3, 2018
+ */
+
+#include <stdint.h>
+#include <stdio.h>
+#include <float.h>
+
+#include "arith128.h"
+#include <testsuite/arith128_print.h>
+#include "vec_int64_ppc.h"
+#include "vec_f64_ppc.h"
+#include <testsuite/arith128_test_f64.h>
+#include <testsuite/vec_perf_f64.h>
+
+#define __DOUBLE_INF (0x7ff0000000000000UL)
+#define __DOUBLE_NINF (0xfff0000000000000UL)
+#define __DOUBLE_NAN (0x7ff0000000000001UL)
+#define __DOUBLE_NNAN (0xfff0000000000001UL)
+#define __DOUBLE_SNAN (0x7ff8000000000001UL)
+#define __DOUBLE_NSNAN (0xfff8000000000001UL)
+#define __DOUBLE_TRUE (0xffffffffffffffffUL)
+#define __DOUBLE_NTRUE (0x0000000000000000UL)
+
+
+#ifdef __DEBUG_PRINT__
+static inline vb64_t
+db_vec_isnormalf64 (vf64_t vf64)
+  {
+    vui64_t tmp, tmp2;
+    const vui64_t expmask = CONST_VINT128_DW(0x7ff0000000000000UL, 0x7ff0000000000000UL);
+    const vui64_t vec_denorm = CONST_VINT128_DW(0x0010000000000000UL, 0x0010000000000000UL);
+    vb64_t result;
+
+    print_v2f64x ("db_vec_isnormalf64:", vf64);
+
+#if _ARCH_PWR7
+    /* Eliminate const load. */
+    tmp2 = (vui64_t)vec_abs (vf64);
+#else
+    const vui64_t signmask = CONST_VINT128_DW(0x8000000000000000UL, 0x8000000000000000UL);
+    tmp2 = vec_andc ((vui64_t)vf64, signmask);
+#endif
+    print_vint128x ("              tmp2=", (vui128_t)tmp2);
+    tmp = vec_and ((vui64_t)vf64, expmask);
+    print_vint128x ("              tmp =", (vui128_t)tmp);
+    tmp2 = (vui64_t)vec_cmplt(tmp2, vec_denorm);
+    print_vint128x ("              tmp2=", (vui128_t)tmp2);
+    tmp = (vui64_t)vec_cmpeq(tmp, expmask);
+    print_vint128x ("              tmp =", (vui128_t)tmp);
+    result = (vb64_t)vec_nor (tmp, tmp2);
+    print_v2b64x ("            result=", result);
+
+    return (result);
+  }
+#endif
+
+int
+test_double_abs (void)
+{
+  vf64_t i, e;
+  vf64_t k;
+  int rc = 0;
+
+  printf ("\n%s double absolute value\n", __FUNCTION__);
+
+  i = (vf64_t) { 0.0, -0.0};
+  e = (vf64_t) { 0.0, 0.0};
+  k = vec_absf64 (i);
+
+#ifdef __DEBUG_PRINT__
+  print_v2f64x ("vec_absf64 i=", i);
+  print_v2f64x ("           k=", k);
+#endif
+  rc += check_v2f64x ("vec_absf64 1:", k, e);
+
+  i = (vf64_t) { -(__DBL_MAX__), __DBL_MIN__ };
+  e = (vf64_t) { __DBL_MAX__, __DBL_MIN__ };
+  k = vec_absf64 (i);
+
+#ifdef __DEBUG_PRINT__
+  print_v2f64x ("vec_absf64 i=", i);
+  print_v2f64x ("           k=", k);
+#endif
+  rc += check_v2f64x ("vec_absf64 1:", k, e);
```

Comment:
I believe the `1:` should be a `2:` to distinguish this from the previous stanza.  This is true for all of the following `check_v2f64x` calls in this function, that the index needs to be updated to `3:`, `4:`, ...